axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderLineServiceImpl.java

```diff
@@ -429,4 +432,42 @@ public void checkMultipleQty(SaleOrderLine saleOrderLine, ActionResponse respons
         product.getAllowToForceSaleQty(),
         response);
   }
+
+  @Override
+  public List<SaleOrderLine> createPackLines(Product product, SaleOrder saleOrder)
+      throws AxelorException {
+    List<SaleOrderLine> subLines = new ArrayList<>();
+
+    for (PackLine packLine : product.getPackLines()) {
+      SaleOrderLine subLine = new SaleOrderLine();
+      Product subProduct = packLine.getProduct();
+      subLine.setProduct(subProduct);
+      subLine.setProductName(subProduct.getName());
+      subLine.setPrice(subProduct.getSalePrice());
+      subLine.setUnit(this.getSaleUnit(subLine));
+      subLine.setQty(new BigDecimal(packLine.getQuantity()));
+      subLine.setCompanyCostPrice(this.getCompanyCostPrice(saleOrder, subLine));
+      TaxLine taxLine = this.getTaxLine(saleOrder, subLine);
+      subLine.setTaxLine(taxLine);
+
+      this.computeValues(saleOrder, subLine);
+
+      BigDecimal price = this.getUnitPrice(saleOrder, subLine, taxLine);
+
+      Map<String, Object> discounts = this.getDiscount(saleOrder, subLine, price);
+
+      if (discounts != null) {
+        subLine.setDiscountAmount((BigDecimal) discounts.get("discountAmount"));
+        subLine.setDiscountTypeSelect((Integer) discounts.get("discountTypeSelect"));
+        if (discounts.get("price") != null) {
+          price = (BigDecimal) discounts.get("price");
+        }
+      }
+      subLine.setPrice(price);
+
+      subLines.add(subLine);
+    }
+
+    return subLines;
+  }
```

Comment:
Please, add a new method to create a saleOrderLine in order to override it in supplychain module to fill the specific fields, instead of override completely the parent method.